science/DMFTtools/Portfile

```diff
@@ -0,0 +1,53 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           cmake 1.1
+PortGroup           compilers 1.0
+PortGroup           github 1.0
+
+github.setup        QcmPlab DMFTtools 2.6.4
+revision            0
+categories          science fortran
+license             LGPL-3
+maintainers         {@barracuda156 gmail.com:vital.had} openmaintainer
+description         A collection of Fortran modules and routines to support quantum many-body calculations, \
+                    with a strong focus on Dynamical Mean-Field Theory
+long_description    {*}${description}
+checksums           rmd160  2a70f56c6a258d0ff21dacbcfc5054e555db218d \
+                    sha256  6515010580fd53f8b82f66c686f0423da1aa41ed8c26a33c63994988554fbec7 \
+                    size    190940
+github.tarball_from archive
+
+cmake.generator     Ninja
+
+depends_build-append \
+                    port:pkgconfig
+depends_lib-append  port:scifor
+
+patchfiles          patch-prefix.diff \
+                    patch-dmft_tools.pc.diff
+
+post-patch {
+    reinplace "s,@PREFIX@,${prefix}," ${worksrcpath}/cmake/MainConfig.cmake
+    reinplace "s,@PREFIX@,${prefix}," ${worksrcpath}/etc/dmft_tools.pc.in
+    reinplace "s,@VERSION@,${version}," ${worksrcpath}/etc/dmft_tools.pc.in
+}
```

Comment:
why can you not set `-DCMAKE_INSTALL_PREFIX` instead?

Reply:
Because it does not work correctly. Removing all patching and setting `-DCMAKE_INSTALL_PREFIX=${prefix}`, it configures to:
```
-- Version file: /opt/local/var/macports/build/_opt_PPCRosettaPorts_science_DMFTtools/DMFTtools/work/build/etc/modules/dmft_tools/gnu/.version
-- Module file: dmft_tools/gnu/2.2.3
-- Library version: 2.2.3
-- Library will be installed in: /opt/local/var/macports/build/_opt_PPCRosettaPorts_science_DMFTtools/DMFTtools/work/.home/opt/dmft_tools/gnu/2.2.3
```
Which ends in a disaster, including these insane paths written into pkgconfig file etc.
```
-- Installing: /opt/local/var/macports/build/_opt_PPCRosettaPorts_science_DMFTtools/DMFTtools/work/destroot/opt/local/var/macports/build/_opt_PPCRosettaPorts_science_DMFTtools/DMFTtools/work/.home/opt/dmft_tools/gnu/2.2.3/version
CMake Error at cmake_install.cmake:129 (file):
  file INSTALL cannot find
  "/opt/local/var/macports/build/_opt_PPCRosettaPorts_science_DMFTtools/DMFTtools/work/.home/opt/dmft_tools/gnu/2.2.3/etc/dmft_tools.pc":
  No such file or directory.


FAILED: CMakeFiles/install.util 
cd /opt/local/var/macports/build/_opt_PPCRosettaPorts_science_DMFTtools/DMFTtools/work/build && /opt/local/bin/cmake -P cmake_install.cmake
ninja: build stopped: subcommand failed.
Command failed:  cd "/opt/local/var/macports/build/_opt_PPCRosettaPorts_science_DMFTtools/DMFTtools/work/build" && ninja install 
Exit code: 1
Error: Failed to destroot DMFTtools: command execution failed
Error: See /opt/local/var/macports/logs/_opt_PPCRosettaPorts_science_DMFTtools/DMFTtools/main.log for details.
Error: Follow https://guide.macports.org/#project.tickets if you believe there
is a bug.
Error: Processing of port DMFTtools failed
```

Reply:
that sounds like an upstream problem, so please report it there.

Reply:
@reneeotten Opened an issue: https://github.com/QcmPlab/DMFTtools/issues/2